api/registrations/serializers.py

```diff
@@ -231,13 +233,36 @@ def create(self, validated_data):
         registration_choice = validated_data.pop('registration_choice', 'immediate')
         embargo_lifted = validated_data.pop('lift_embargo', None)
         reviewer = is_prereg_admin_not_project_admin(self.context['request'], draft)
+        children = validated_data.pop('children', None)
+        if children:
+            parent_node = draft.branched_from
+
+            # Validate to make sure we aren't registering nodes without registering their parents
+            # and all nodes have the same root.
+
+            # First check that all children are valid
+            child_nodes = Node.objects.filter(guids___id__in=children)
+            if child_nodes.count() != len(children):
+                    raise exceptions.ValidationError('Some child nodes could not be found.')
+
+            # Second check that all children have a parent being registered. The exception being
+            #  the root node of the registration.
+            for node in child_nodes:
+                if node.parent_node and node.parent_node not in list(child_nodes) + [parent_node] and node != parent_node:
```

Comment:
I'm a bit worried about the number of queries being added here: (2 `.parent_node` JOINs + 1 `list(child_nodes)` select) * n nodes + 1 query for looping over child_nodes == 3n+1.

Can you test this out with a moderately-large hierarchy and make sure it wouldn't time out?

One simple optimization would be to cache the queries by saving the results to variables

```python
child_nodes_set = set(child_nodes)
for node in child_nodes:
    parent_node = node.parent_node
    if parent_node and parent_node not in child_nodes_set ...
```

Still n+1, though, so that might not suffice. Can you optimize this further?